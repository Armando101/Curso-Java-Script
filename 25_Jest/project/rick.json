{
	"id": 1,
	"name": "Rick Sanchez",
	"status": "Alive",
	"species": "Human",
	"gender": "Male"
}