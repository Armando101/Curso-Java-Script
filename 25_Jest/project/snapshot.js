export const getCharacter = (data) => ({
	 
	"id": data.id,
	"name": data.name,
	"status": data.status,
	"species": data.species,
	"gender": data.gender
	
});