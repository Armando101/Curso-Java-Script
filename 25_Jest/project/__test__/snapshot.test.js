import { getCharacter } from '../snapshot';
import rick from '../rick.json';

describe('Es hora de las instantaneas', () => {
	test('Snapshot', () => {
		expect(getCharacter(rick)).toMatchSnapshot();
	});

	test('Siempre fallara la instantanea', () => {
		const user = {
			createAt: new Date(),
			id: Math.floor(Math.random() * 20),
		};
		expect(user).toMatchSnapshot();
	});

	test('Tenemos una excpecion', ()=> {
		const user = {
			id: Math.floor(Math.random() * 20),
			name: 'Armando Rivera'
		}
		expect(user).toMatchSnapshot({
			id: expect.any(Number) 	// Indico que no importa el id en la comparacion con el snapshot
		});
	});
});