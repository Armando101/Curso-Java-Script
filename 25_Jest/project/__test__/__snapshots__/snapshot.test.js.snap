// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Es hora de las instantaneas Siempre fallara la instantanea 1`] = `
Object {
  "createAt": 2020-05-29T02:20:13.272Z,
  "id": 11,
}
`;

exports[`Es hora de las instantaneas Snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanchez",
  "species": "Human",
  "status": "Alive",
}
`;

exports[`Es hora de las instantaneas Tenemos una excpecion 1`] = `
Object {
  "id": Any<Number>,
  "name": "Armando Rivera",
}
`;
